packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -214,11 +208,64 @@ export enum ContainerMessageType {
 	IdAllocation = "idAllocation",
 }
 
+/**
+ * How should an older client handle an unrecognized remote op type?
+ *
+ * @internal
+ */
+export type CompatModeBehavior =
+	/** Ignore the op. It won't be persisted if this client summarizes */
+	| "Ignore"
+	/** Fail processing immediately. (The container will close) */
+	| "FailToProcess";
+
+/**
+ * All the info an older client would need to know how to handle an unrecognized remote op type
+ *
+ * @internal
+ */
+export interface IContainerRuntimeMessageCompatDetails {
+	/** How should an older client handle an unrecognized remote op type? */
+	behavior: CompatModeBehavior;
+}
+
+/**
+ * Utility to implement compat behaviors given an unknown message type
+ * The parameters are typed to support compile-time enforcement of handling all known types/behaviors
+ *
+ * @param _unknownContainerRuntimeMessageType - Typed as never, to ensure all known types have been
+ * handled before calling this function (e.g. in a switch statement).
+ * @param compatBehavior - Typed redundantly with CompatModeBehavior to ensure handling is added when updating that type
```

Comment:
Not sure how I feel about this, but probably fine to revisit as other types are added

Reply:
Here's the reason not to use the type here: The logic right now only calls out "Ignore"; the behavior for "FailToProcess" is implicit.  If another behavior is added, and this function is overlooked (unlikely, I know, since it's kind of the main point, but still) then the new option will default to the same behavior.  By explicitly calling out here what individual behaviors this function handles, if the other type changes it will force the author to deal with it, because the types will no longer be compatible.

It's similar to putting `unreachableCase` in the `default` case of a `switch` block that covers all known inputs.